File: spring-web-java/src/main/java/spring/web/java/dto/request/MemberSaveRequest.java
package spring.web.java.dto.request;

import jakarta.validation.constraints.NotNull;
import spring.web.java.common.constant.Role;

public record MemberSaveRequest(
        @NotNull
        String account,
        @NotNull
        String password,
        @NotNull
        String name,
        @NotNull
        Role role,
        @NotNull
        String city,
        @NotNull
        String street,
        @NotNull
        String zipcode
) {
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/dto/request/MemberSaveRequest.kt
package spring.web.kotlin.dto.request

import spring.web.kotlin.constant.Role

data class MemberSaveRequest(
    val account: String,
    val password: String,
    val name: String,
    val role: Role,
    val city: String,
    val street: String,
    val zipcode: String
)
